clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -110,7 +110,9 @@ public synchronized Cluster fetch() {
      * will be reset on the next update.
      */
     public synchronized void add(String topic) {
-        topics.put(topic, TOPIC_EXPIRY_NEEDS_UPDATE);
+        if (topics.put(topic, TOPIC_EXPIRY_NEEDS_UPDATE) == null) {
```

Comment:
Good catch! Yeah, I forgot that we should always request a new metadata update when changing the regex subscription, even though it doesn't change anything about the topics we are fetching metadata for.